base/applications/taskmgr/perfdata.c

```diff
@@ -247,15 +255,25 @@ void PerfDataRefresh(void)
     /*
      * Save system processor time info
      */
+#ifdef MAKE_LEAK
     if (SystemProcessorTimeInfo) {
         HeapFree(GetProcessHeap(), 0, SystemProcessorTimeInfo);
     }
     SystemProcessorTimeInfo = SysProcessorTimeInfo;
+#else
+    if (SystemBasicInfo.NumberOfProcessors <= 32)
+        memcpy(&SystemProcessorTimeInfo, &SysProcessorTimeInfo,
+               sizeof(SystemProcessorTimeInfo) * SystemBasicInfo.NumberOfProcessors);
+#endif
 
     /*
      * Save system handle info
      */
     SystemNumberOfHandles = SysHandleInfoData.NumberOfHandles;
+#ifndef MAKE_LEAK
+    // HACK: We need to do this twice or we leak memory
```

Comment:
That's super-weird: how is it possible that just setting twice with the same value, the same global variable, magically "fixes" a memory leak? I would suspect that it relates to some locking problem or something like that.
Could you recheck again that indeed you get that mem leak if you don't set `SystemNumberOfHandles` twice? (but you keep the other SystemProcessorTimeInfo code modifications above)

Reply:
I have found another workaround by just moving this code up a bit, so it does not seem out-of-place now at all.
Now there is no duplication of the same line needed anymore. Thanks.

Reply:
Now I have restored the original code. Thanks.